### PR TITLE
Feature/payment receipt support

### PR DESCRIPTION
**Pull request by BrianCorbin**
Adding support for generating receiver receipts and transaction log confirmations.

### CODE REVIEW

File: app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import * as Yup from 'yup';
+
+import { SubmitButton } from '..';
+import { ConfirmDeleteWalletDialogProps } from './ConfirmDeleteWalletDialog';
+
+const ConfirmDeleteWalletDialog = (props: ConfirmDeleteWalletDialogProps): JSX.Element => {
+  const { open, cancel, confirm } = props;
+  const { t } = useTranslation('TermsOfUseDialog');
+
+  const handleOnSubmit = () => confirm();
```
Comment (fktt22) [inline app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx:16]: Better:

`const handleOnSubmit = confirm;`

File: app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <Dialog
+      open={open}
+      onClose={() => {
+        cancel();
+      }}
```
Comment (fktt22) [inline app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx:23]: Similarly:

`onClose = { cancel }`

File: app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    >
+      <Container maxWidth="md" style={{ marginBottom: '20px', marginTop: '20px' }}>
+        <Typography variant="h2" paragraph>
+          Delete Wallet
+        </Typography>
+        <Typography variant="body2" color="textSecondary" paragraph>
+          If you don't have your entropy backed up, you will lose your account forever!
+        </Typography>
```
Comment (fktt22) [inline app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx:31]: Texts must be from i18n files

File: app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        >
+          {({ errors, isSubmitting, dirty, isValid, setFieldValue, submitForm }) => (
+            <Form name="UnlockWalletInnerForm">
+              <Box display="flex">
+                <Box display="flex" alignItems="center" flexDirection="row-reverse">
+                  <Box>
+                    <Typography display="inline">
+                      I understand this could cause me to lose my accounts if I don't have them
+                      backed up.
+                    </Typography>
```
Comment (fktt22) [inline app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx:47]: Texts must be from i18n files

File: app/pages/Auth/UnlockWallet.view/UnlockWallet.view.tsx [hunk 5]
```diff
@@ -75,6 +78,18 @@ const UnlockWalletView: FC<UnlockWalletViewProps> = ({
             >
               {t('unlockWalletButton')}
             </SubmitButton>
+            <Button
+              color="primary"
+              onClick={() => setConfirmDeleteWalletDialogOpen(true)}
+              id="deleteWalletConfirmation"
+            >
+              Delete Wallet and Start Over
+            </Button>
```
Comment (fktt22) [inline app/pages/Auth/UnlockWallet.view/UnlockWallet.view.tsx:87]: All texts must come from i18n files

File: app/pages/History/TransactionDetails.view/TransactionDetails.view.tsx [hunk 6]
```diff
@@ -145,54 +204,15 @@ const TransactionDetails: FC<TransactionDetailsViewProps> = ({
...
-        </Container> */}
+      {direction === 'tx_direction_sent' ? (
+        <Button id="copyConfirmations" onClick={onClickCopyConfirmations}>
+          <Typography color="textPrimary">Copy Confirmation(s)</Typography>
+        </Button>
+      ) : (
+        <Button id="validateConfirmations" onClick={onClickValidateConfirmations}>
+          <Typography color="textPrimary">Validate Confirmation(s)</Typography>
+        </Button>
+      )}
```
Comment (fktt22) [inline app/pages/History/TransactionDetails.view/TransactionDetails.view.tsx:215]: Texts must be from i18n files

**Review by fktt22 — CHANGES_REQUESTED** [review]
Please review comments, in particular about i18n

File: app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx (hunk 3 above)
Comment (BrianCorbin) [reply app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx:24]: addressed in next PR

File: app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx (hunk 4 above)
Comment (BrianCorbin) [reply app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx:40]: addressed in next PR

File: app/pages/Auth/UnlockWallet.view/UnlockWallet.view.tsx (hunk 5 above)
Comment (BrianCorbin) [reply app/pages/Auth/UnlockWallet.view/UnlockWallet.view.tsx:87]: addressed in next PR

File: app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+          onSubmit={confirm}
+        >
+          {({ errors, isSubmitting, dirty, isValid, setFieldValue, submitForm }) => (
+            <Form name="UnlockWalletInnerForm">
+              <Box display="flex">
+                <Box display="flex" alignItems="center" flexDirection="row-reverse">
+                  <Box>
+                    <Typography display="inline">
+                      I understand this could cause me to lose my accounts if I don't have them
+                      backed up.
```
Comment (fktt22) [inline app/components/ConfirmDeleteWallet.dialog/ConfirmDeleteWalletDialog.view.tsx:39]: This needs translation from i18n

**Review by fktt22 — APPROVED** [review]

**Pull request merged**